modules/build/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -334,21 +342,85 @@ final case class BuildOptions(
             Right(validVersions.max.repr)
       }
     }
-    val maybeSv = scalaVersion match {
+    val maybeSv = scalaVersionArg match {
       case None => matchNewestScalaVersion(None)
       case Some(sv0) =>
         if (Util.isFullScalaVersion(sv0)) Right(sv0)
         else matchNewestScalaVersion(Some(sv0))
     }
 
-    val sv  = value(maybeSv)
-    val sbv = scalaBinaryVersion.getOrElse(ScalaVersion.binary(sv))
-    (sv, sbv)
+    val scalaVersion       = value(maybeSv)
+    val scalaBinaryVersion = scalaBinaryVersionArg.getOrElse(ScalaVersion.binary(scalaVersion))
+    (scalaVersion, scalaBinaryVersion)
   }
 
+  /** @return
+    *   Either a BuildException or the calculated (ScalaVersion, ScalaBinaryVersion) tuple
+    */
+  private def computeLatestScalaThreeNightlyVersions(): Either[BuildException, (String, String)] =
+    either {
+      import coursier.Versions
+      import coursier.core.Latest
+      import coursier._
+      import scala.concurrent.ExecutionContext.{global => ec}
+      val moduleVersion: Option[String] = {
+        def scala3 = mod"org.scala-lang:scala3-library_3"
+        val res = finalCache.logger.use {
+          Versions()
+            .withModule(scala3)
+            .result()
+            .unsafeRun()(ec)
+        }
+        res.versions.latest(Latest.Release)
+      }
+      val scalaVersion       = moduleVersion.get
+      val scalaBinaryVersion = ScalaVersion.binary(scalaVersion)
+      (scalaVersion, scalaBinaryVersion)
+    }
+
+  /** @return
+    *   Either a BuildException or the calculated (ScalaVersion, ScalaBinaryVersion) tuple
+    */
+  private def computeLatestScalaTwoNightlyVersions(): Either[BuildException, (String, String)] =
+    either {
+      import coursier.Versions
+      import coursier.core.Latest
+      import coursier._
+      import scala.concurrent.ExecutionContext.{global => ec}
+      val scala2NightlyRegex = raw"""(\d+)\.(\d+)\.(\d+)-bin-[a-f0-9]*""".r
+
+      def isScala2Nightly(version: String): Boolean =
+        scala2NightlyRegex.unapplySeq(version).isDefined
+
+      val moduleVersion: Option[String] = {
+        def scalaNightly2Module: Module = mod"org.scala-lang:scala-library"
+        val res = finalCache.logger.use {
+          Versions()
+            .withModule(scalaNightly2Module)
+            .withRepositories(Seq(coursier.Repositories.scalaIntegration))
+            .result()
+            .unsafeRun()(ec)
+        }
+        res.versions.latest(Latest.Release)
+      }
+      val scalaVersion       = moduleVersion.get
```

Comment:
Same as above, no `Option.get`.

Reply:
I have solved this in my scala-version-error commit which is in draft stage.

Reply:
I return `Either` from the `moduleVersion`, then use `value` for extracting it.

Reply:
These functions are changed quite a bit in the scala-version-error PR

Reply:
I am thinking about closing this PR and only keeping the scala-version-error, since the same functions are modified in there.

Reply:
I think we should open PR per one issue. It's easier to make a code review.

Reply:
It makes a lot of sense.